pageserver/src/lib.rs

```diff
@@ -85,6 +87,50 @@ pub async fn shutdown_pageserver(
     exit_code: i32,
 ) {
     use std::time::Duration;
+
+    let walredo_extraordinary_shutdown_thread_span = {
+        let span = info_span!(parent: None, "walredo_extraordinary_shutdown_thread");
+        span.follows_from(tracing::Span::current());
+        span
+    };
+    let walredo_extraordinary_shutdown_thread = std::thread::spawn(move || {
```

Comment:
why is it a thread and not something spawned in a tokio runtime? I suppose it's actually cheaper to bring up a thread than an entire runtime (which has multiple threads), but a comment might be nice why the choice fell on a thread (threads are processes, so in order to kill extra processes we make a new process).

It's an okay choice, but a comment to explain it would be nice.

Reply:
52d7b3882ba8a125f40b4db3031129c7406931a1